app/controllers/admin/submissions_controller.rb

```diff
@@ -1,11 +1,85 @@
 # frozen_string_literal: true
 
+class StateActions
+  def self.for(submission)
+    new(submission).run
+  end
+
+  attr_reader :submission
+
+  def initialize(submission)
+    @submission = submission
+  end
+
+  def run
+    actions = []
+    actions << approve_action if submission.submitted?
+    actions << publish_action if submission.submitted? || submission.approved?
+    actions << reject_action if submission.submitted?
+    actions << close_action if submission.submitted?
+    actions
+  end
+
+  private
+
+  def default_classes
+    ['btn btn-secondary btn-small btn-full-width']
+  end
+
+  def approve_action
+    {
+      class: default_classes << 'btn-approve',
+      confirm:
+        'An email will be sent to the consignor, letting them know that their submission will be sent to our partner network and this work will appear in the digests and CMS. This action cannot be undone.',
```

Comment:
I think this message is inaccurate now, as we no longer send emails for the `Approve` action. I think we could reuse the message from the old `Approve without digest` action: 

```suggestion
        'No email will be sent to the consignor and this submission will be excluded from the digests.',
```

Reply:
Fixed with 40266b5.